pkg/webhooks/cluster_webhook.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright Kurator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/cluster-api/util/version"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	v1 "kurator.dev/kurator/pkg/apis/cluster/v1alpha1"
+)
+
+const (
+	defaultPodCIDR     = "192.168.0.0/16"
+	defaultServiceCIDR = "10.96.0.0/12"
+	defaultVpcCIDR     = "10.0.0.0/16"
```

Comment:
document it in api

Reply:
it's done. Consider moving to `pkg/apis` as constants?
https://github.com/kurator-dev/kurator/blob/7516fb3f4cb168cda9d5119d930ddb32a893988d/pkg/apis/cluster/v1alpha1/cluster_types.go#L90

Reply:
IMO not necessary